common/test/kotlin/com/couchbase/lite/DeprecatedConfigFactoryTest.kt

```diff
@@ -0,0 +1,130 @@
+    }
+
+    // Create on factory with no db should fail
+    @Test(expected = IllegalArgumentException::class)
+    fun testReplicatorConfigNoDb() {
+        ReplicatorConfigurationFactory.create(target = testEndpoint, type = ReplicatorType.PULL)
+    }
+
+    // Create on factory with no target should fail
+    @Test(expected = IllegalArgumentException::class)
+    fun testReplicatorConfigNoProtocol() {
+        ReplicatorConfigurationFactory.create(testDatabase, type = ReplicatorType.PULL)
+    }
+
+    // Create on factory with db with no default collection should fail
+    @Test(expected = IllegalArgumentException::class)
+    fun testReplicatorConfigNoDefaultCollection() {
+        testDatabase.defaultCollection?.delete()
+        ReplicatorConfigurationFactory.create(testDatabase, testEndpoint)
+    }
+
+    // Create with db and endpoint should succeed
+    @Test
+    fun testReplicatorConfigWithGoodArgs() {
+        val config = ReplicatorConfigurationFactory.create(testDatabase, testEndpoint)
+        assertEquals(testDatabase, config.database)
+        assertEquals(testEndpoint, config.target)
+    }
+
+    // Create should copy source
+    @Test
+    fun testReplicatorConfigCopy() {
+        val config1 = ReplicatorConfigurationFactory.create(testDatabase, testEndpoint, type = ReplicatorType.PULL)
+        val config2 = config1.create()
+        assertNotSame(config1, config2)
+        assertEquals(config1.database, config2.database)
+        assertEquals(config1.target, config2.target)
+        assertEquals(config1.type, config2.type)
+    }
+
+    // Create should replace source
+    @Test
+    fun testReplicatorConfigReplace() {
+        val config1 = ReplicatorConfigurationFactory.create(testDatabase, testEndpoint, type = ReplicatorType.PULL)
+        val config2 = config1.create(type = ReplicatorType.PUSH)
+        assertNotSame(config1, config2)
+        assertEquals(config1.database, config2.database)
+        assertEquals(config1.target, config2.target)
+        assertEquals(ReplicatorType.PUSH, config2.type)
+    }
+
+    // Create from a source explicitly specifying a default collection
+    @Test
+    fun testReplicatorConfigFromCollectionWithDefault() {
+        val config1 = ReplicatorConfigurationFactory
+            .newConfig(testEndpoint, mapOf(testDatabase.defaultCollection!! to CollectionConfiguration()))
+        val config2 = config1.create()
+        assertNotSame(config1, config2)
+        assertEquals(config1.database, config2.database)
+        assertEquals(setOf(testCollection.database.defaultCollection), config2.collections)
+    }
+
+    // Create from a source with default collection, explicitly specifying a non-default collection
+    @Test
+    fun testReplicatorConfigFromCollectionWithDefaultAndOther() {
+        val config1 = ReplicatorConfigurationFactory
+            .newConfig(testEndpoint, mapOf(testCollection to CollectionConfiguration()))
+        val config2 = config1.create()
```

Comment:
Will config2 has the same collection config (which has testCollection) as the config1?

Reply:
Not quite.  ReplicatorConfiguration.create is deprecated.  It cannot handle arbitrary collections.  If the source replicator configuration contains collections other than the default collection, it does not know how to copy them.

This situation is easily avoided by not mixing ReplicatorConfiguration.create with ReplicatorConfiguration.newConfig (the latter being the new, collection enabled, signature)